enterprise/j2ee.core/src/org/netbeans/api/j2ee/core/Profile.java

```diff
@@ -140,134 +123,23 @@ public String toString() {
-        boolean isParamFullProfile = isFullProfile(comparingVersion);
-        if (isThisFullProfile && isParamFullProfile) {
-            // Both profiles are Java EE Full
-            return true;
-        }
-        if (!isThisFullProfile && !isParamFullProfile) {
-            // Both profiles are Java EE Web
-            return true;
-        }
-        if (isThisFullProfile && !isParamFullProfile) {
-            // profileToCompare is Java EE Full profile and comparingVersion is only Java EEWeb profile
-            return true;
-        }
-        return false;
-    }
-
-    private static boolean isFullProfile(@NonNull Profile profile) {
-        final String profileDetails = profile.toPropertiesString();
-        if (profileDetails.indexOf("-") == -1) {
-            return true;
+        String valueMinusQuotes = value;
+        if(value.contains("\"")){
+            valueMinusQuotes = value.replace("\"","");
         }
-        return false;
-    }
 
-    /**
-     * Compares if the first given profile has equal or higher Java EE version
-     * in comparison to the second profile.
-     *
-     * Please be aware of the following rules:
-     * <br/><br/>
-     *
-     * 1) Each Java EE X version is considered as lower than Java EE X+1 version
-     * (this applies regardless on Web/Full specification and in reality it means
-     * that even Java EE 6 Full version is considered as lower than Java EE 7 Web)
-     * <br/><br/>
-     *
-     * 2) Each Java EE X Web version is considered as lower than Java EE X Full
-     * <br/>
-     *
-     * @param profileToCompare profile that we want to compare
-     * @param comparingVersion version which we are comparing with
-     * @return <code>true</code> if the profile version is equal or higher in
-     *         comparison with the second one, <code>false</code> otherwise
-     * @since 1.19
-     */
-    private static boolean isVersionEqualOrHigher(@NonNull Profile profileToCompare, @NonNull Profile comparingVersion) {
-        int comparisonResult = Profile.UI_COMPARATOR.compare(profileToCompare, comparingVersion);
-        if (comparisonResult == 0) {
-            // The same version for both
-            return true;
-        } else {
-            String profileToCompareVersion = getProfileVersion(profileToCompare);
-            String comparingProfileVersion = getProfileVersion(comparingVersion);
-            if (profileToCompareVersion.equals(comparingProfileVersion)) {
-                return compareWebAndFull(profileToCompare, comparingVersion);
-            } else {
-                // profileToCompare has lower version than comparingVersion
-                return comparisonResult <= 0;
+        for (Profile profile : Profile.values()) {
+            if (profile.toPropertiesString().equals(valueMinusQuotes)
+                    || profile.name().equals(value)
+                    || profile.name().endsWith(value)) {
+                return profile;
             }
         }
```

Comment:
this looks a little bit suspicious. The `equals` in the middle is redundant right now since it is followed by an `endsWith`. If this is intended, it is probably still better to remove the equals to make it look less suspicious :)

Shouldn't it use `valueMinusQuotes` everywhere?

Reply:
I've done this on purpose to stick to existing logic.

But I guess we can use `valueMinusQuotes` for all statements.

Reply:
> I've done this on purpose to stick to previous logic.

So where does the `endsWith` come from?

Reply:
It represents the additional checks coming from JAVA_EE_* type checks.

```java
"EE_6_FULL".equals(value))
[...]
"EE_6_WEB".equals(value))
...
```

I know it's not 100% the same as before, but when passing `EE_*_FULL` or `EE_*_WEB` it should behave like before.

Reply:
I think I get it: So the input can be `"6-web"` or `"EE_6_WEB" `and you match it against the postfix of the enum name `JAVA_EE_6_WEB`

how about:
```java
        for (Profile profile : Profile.values()) {
            if (profile.toPropertiesString().equals(valueMinusQuotes)
                    || profile.name().equals(valueMinusQuotes)
                    || (valueMinusQuotes.startsWith("EE_") && profile.name().endsWith(valueMinusQuotes))) {
                return profile;
            }
        }
```
this is likely less risky and does also communicate a bit better what is going on there.

To me, a `equals` followed by `endsWith` looks a bit suspicious since it makes the equals redundant. It looks like a bug.